StRoot/StRefMultCorr/StBadRunChecker.cxx

```diff
@@ -0,0 +1,317 @@
+}
+
+//______________________________________________________________________________
+void StBadRunChecker::readBadRunsFromHeaderFile() {
+    switch(mRunIndex){
+        case 0:
+            for(int i=0;i<2;i++){
+                mRunRange.push_back(run19_col_19p6_auau_range[i]);
+            }
+            for(int i=0;i<nBadRun_run19_col_19p6_auau;i++){
+                mBadRun_all.push_back(badrun_run19_col_19p6_auau_all[i]);
+                std::vector<Int_t> row;
+                for(int j=0;j<12;j++){
+                    row.push_back(badrun_run19_col_19p6_auau_sub[i][j]);       
+                }
+                mBadRun_sub.push_back(row);
+            } 
+            cout<<"read in bad run list for "<<mRun<<" "<<mColMode<<" "<<mSpecies<<" "<<mEnergy<<" GeV. "<<nBadRun_run19_col_19p6_auau<<" bad runs in total."<<endl;
+            break;
+        /*
+        case 1:
+            for(int i=0;i<2;i++){
+                mRunRange.push_back(run19_col_14p6_auau_range[i]);
+            }
+            for(int i=0;i<nBadRun_run19_col_14p6_auau;i++){
+                mBadRun_all.push_back(badrun_run19_col_14p6_auau_all[i]);
+                std::vector<Int_t> row;
+                for(int j=0;j<12;j++){
+                    row.push_back(badrun_run19_col_14p6_auau_sub[i][j]);       
+                }
+                mBadRun_sub.push_back(row);
+            } 
+            cout<<"read in bad run list for "<<mRun<<" "<<mColMode<<" "<<mSpecies<<" "<<mEnergy<<" GeV. "<<nBadRun_run19_col_14p6_auau<<" bad runs in total."<<endl;
+            break;
+        */
+        //Add more runs here
+        default:
+            cout<<mRun<<" "<<mColMode<<" "<<mSpecies<<" "<<mEnergy<<" GeV doesn't exist or the bad run list is not available yet.Exit!"<<endl;
+            exit(0);
+    }
+
+}
+
+
+Bool_t StBadRunChecker::isBadRun(const Int_t RunId,TString mS){
+    // Return true if a given run id is bad run
+    if (RunId<mRunRange[0]||RunId>mRunRange[1]){
+        cout<<"Warning: "<<RunId<<" is not in "<<mRun<<" "<<mColMode<<" "<<mSpecies<<" "<<mEnergy<<"GeV. Exit!"<<endl;
+        exit(0);
+    }
+    mS.ToLower(); 
+    vector<Int_t>::iterator iter = std::find(mBadRun_all.begin(), mBadRun_all.end(), RunId);
+    if(iter == mBadRun_all.end()){
+        return 0;//RunId is not a bad run 
+    }
+    //return ( iter != mBadRun.end() ) ;
+    else{
+        int index = std::distance(mBadRun_all.begin(), iter);//index starts at zero
+        int isbad=1;
+        for(int iname=0;iname<12;iname++){
+            mSubSysName[iname].ToLower();
+            //As long as this run was marked as a bad run for one of the sub systems, it is a bad run
+            if(mS.Contains(mSubSysName[iname])){
+                cout<<"Check if this run is bad for: "<<mSubSysName[iname]<<endl;
+                isbad*=(1-mBadRun_sub[index][iname]);
+            }
+        }
+        return (isbad==0);
+    }
+}
+
+Bool_t StBadRunChecker::isInjection(const Int_t RunId){
```

Comment:
Instead of implementing new codes, could you just call "isBadRun(RunId, Injection)"? Same comments for all other subsystem-specific functions.

Reply:
Yes, you can just call isBadRun(RunId,injection). The subsystem-specific functions are indeed optional. I was thinking maybe some users prefer calling methods with explicit names instead of passing arguments to a method themselves, so they don't need to worry about typos? It doesn't hurt to keep both but I am also okay with removing all the subsystem-specific functions if you think it will cause confusion.

Reply:
I think it is a good idea to have those subsystem-specific functions. My comment was that you can use the master isBadRun() function for implementation, instead of writing somewhat duplicated codes. Something like: 
Bool_t StBadRunChecker::isInjection(const Int_t RunId){
return isBadRun(RunId,injection);
}